Libs/Widgets/ctkLanguageComboBox.cpp

```diff
@@ -35,7 +35,17 @@ class ctkLanguageComboBoxPrivate
 public:
   ctkLanguageComboBoxPrivate(ctkLanguageComboBox& object);
   void init();
+
   void updateLanguageItems();
+
+  /// \brief Normalize Locale Code
+  ///
+  /// Converts the input locale code to its long form, handling incomplete or
+  /// extended inputs. For example, "en" -> "en_US", "fr" -> "fr_FR", "fr_" -> "fr_FR",
+  /// "en_U" -> "en_US", "fr_FRRRR" -> "fr_FR". If the input is invalid, it returns
+  /// an empty string.
+  QString normalizeLocaleCode(const QString& localeCode);
```

Comment:
Why is it good that we accept invalid strings, such as "en_U" or "fr_FRRRR"?

Reply:
This is consistent with the behavior of `qt.QLocal` and only document it.

Reply:
In other word, it is part of the "normalization process" ...

Reply:
I would consider this as a limitation, but since it would be hard to change the Qt implementation, just documenting it is fine. Maybe add a note that this behavior is due to how QLocale is implemented.